src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateStorageSession.cs

```diff
@@ -93,9 +95,9 @@ protected override void Setup(FeatureConfigurationContext context)
                 var persisterFactory = context.Settings.Get<IOutboxPersisterFactory>();
                 var persister = persisterFactory.Create(sessionFactory, context.Settings.EndpointName(), pessimisticMode, transactionScopeMode);
 
-                context.Container.ConfigureComponent(b => persister, DependencyLifecycle.SingleInstance);
-                context.Container.ConfigureComponent(b => new NHibernateSynchronizedStorage(sessionFactory, sessionHolder), DependencyLifecycle.SingleInstance);
-                context.Container.ConfigureComponent(b => new NHibernateSynchronizedStorageAdapter(sessionFactory, sessionHolder), DependencyLifecycle.SingleInstance);
+                context.Services.AddSingleton<IOutboxStorage>(_ => persister);
```

Comment:
Why factory method here, but not on the other two?

Reply:
probably no reason, just missed to make the final refactoring step.